pkg/koordlet/runtimehooks/hooks/cpuset/rule.go

```diff
@@ -54,14 +54,31 @@ func (r *cpusetRule) getContainerCPUSet(containerReq *protocol.ContainerRequest)
 	}
 	podAnnotations := containerReq.PodAnnotations
 	podLabels := containerReq.PodLabels
-	podAlloc, err := ext.GetResourceStatus(podAnnotations)
+	podAlloc, err := extension.GetResourceStatus(podAnnotations)
 	if err != nil {
 		return nil, err
 	}
 
-	podQOSClass := ext.GetQoSClassByAttrs(podLabels, podAnnotations)
-	if len(podAlloc.NUMANodeResources) != 0 {
-		getCPUFromSharePoolByAllocFn := func(sharePools []ext.CPUSharedPool, alloc *ext.ResourceStatus) string {
+	podQOSClass := extension.GetQoSClassByAttrs(podLabels, podAnnotations)
+
+	// SYSTEM QoS cpuset take precedence to sharepool
+	if podQOSClass == extension.QoSSystem && len(r.systemQOSCPUSet) > 0 {
+		klog.V(6).Infof("get cpuset from system qos rule for container %s/%s",
+			containerReq.PodMeta.String(), containerReq.ContainerMeta.Name)
+		return pointer.String(r.systemQOSCPUSet), nil
+	}
+
+	// numa-aware
+	isNUMAAware := false
+	for _, numaNode := range podAlloc.NUMANodeResources {
+		// check if cpu resource is allocated in numa-level since there can be numa allocation without cpu
+		if numaNode.Resources != nil && !numaNode.Resources.Cpu().IsZero() {
+			isNUMAAware = true
+			break
+		}
+	}
+	if isNUMAAware {
```

Comment:
cpu allocate result should be the first selection if scheduler has specified

Reply:
done